# Make sure that we use EasyBuild compiler flags
# SEP 4th 2020 by B. Hajgato (UGent)
--- config/compilers/gcc_defaults.mak.orig	2017-09-04 17:54:03.000000000 +0200
+++ config/compilers/gcc_defaults.mak	2020-09-04 14:51:15.343251060 +0200
@@ -38,8 +38,8 @@
  ##                                                                       ##
  ###########################################################################
 
-CC=gcc
-CXX=gcc
+CC?=gcc
+CXX?=gcc
 ifeq ($(OSTYPE),Darwin)
 CXX=g++
 endif
@@ -47,8 +47,8 @@
 COMPILER_DESC=FSF gcc
 COMPILER_VERSION_COMMAND=$(CXX) -v 2>&1 | tail -1 | sed -e 's/^....//'
 
-CFLAGS  = $(GCC_SYSTEM_OPTIONS) $(CC_OTHER_FLAGS)
-CXXFLAGS  =  $(GCC_SYSTEM_OPTIONS) -fno-implicit-templates $(CC_OTHER_FLAGS) $(OMP_OPTS) $(OMP_DEFS)
+CFLAGS  = $(GCC_SYSTEM_OPTIONS) $(CC_OTHER_FLAGS) ${EBVARCFLAGS}
+CXXFLAGS  =  $(GCC_SYSTEM_OPTIONS) -fno-implicit-templates $(CC_OTHER_FLAGS) $(OMP_OPTS) $(OMP_DEFS) ${EBVARCXXFAGS}
 
 DEBUG_CCFLAGS   = -g
 DEBUG_CXXFLAGS  = -g
@@ -62,9 +62,9 @@
 VERBOSE_CXXFLAGS  = 
 VERBOSE_LINKFLAGS =
 
-OPTIMISE_CCFLAGS   = -O$(OPTIMISE)
-OPTIMISE_CXXFLAGS  = -O$(OPTIMISE)
-OPTIMISE_LINKFLAGS = -O$(OPTIMISE)
+OPTIMISE_CCFLAGS   =  
+OPTIMISE_CXXFLAGS  =  
+OPTIMISE_LINKFLAGS = 
 
 PROFILE_DEFAULT = gprof
 
